x/concentrated-liquidity/internal/math/tick.go

```diff
@@ -50,14 +50,18 @@ func TickToSqrtPrice(tickIndex, exponentAtPriceOne sdk.Int) (price sdk.Dec, err
 	geometricExponentIncrementDistanceInTicks := sdkNineDec.Mul(powTen(exponentAtPriceOne.Neg()))
 
 	// Check that the tick index is between min and max value for the given exponentAtPriceOne
-	minTick, maxTick := GetMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne)
+	minTick, maxTick := getMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne)
 	if tickIndex.LT(sdk.NewInt(minTick)) {
 		return sdk.Dec{}, types.TickIndexMinimumError{MinTick: minTick}
 	}
 	if tickIndex.GT(sdk.NewInt(maxTick)) {
 		return sdk.Dec{}, types.TickIndexMaximumError{MaxTick: maxTick}
 	}
 
+	if tickIndex.Equal(sdk.NewInt(minTick)) {
```

Comment:
Does this mean that price is always zero at min tick?

Reply:
Also noticed that we changed from using minsqrt price -> zero, what is the context for this?

Reply:
This is still up for debate on how we want to treat this, but the logic is as follows:

Whatever the min tick is should be as close to 0 as possible and whatever the max tick is should be as close to inf as possible.

The problem we run into here with the minTick being .000000000001 is the sqrt of this is a larger number (.000001) and results in the user getting less shares than they should when transferring from a 0 to inf balancer position. In reality, the min tick should be 0 and the sqrt of 0 is 0. So this tries to replicate this logic.

Reply:
I see, thanks for the explanation. I think it would be nice to have more intense tests that especially touches this scenario, since it would result in better safeness imo! Doesnt have to be in this specific PR but can be a separate PR

Reply:
Would it also be possible to add in line comments from the context you have mentioned?

Reply:
added in line comment explaining this https://github.com/osmosis-labs/osmosis/pull/4051/commits/ae60111c0c026afa84f24ddbaf66e591ef8cfe9d

Reply:
I know that I've agreed about this change during discussions. However, after looking at it the review, I'm concerned about returning 0 for the square root price. Price cannot be zero so this value is invalid, and I'm worried that it will cause us unforeseen problems and bugs in the future.

I prefer giving out fewer shares to users for now while preserving the current model with its correctness.

If you disagree, let's not block this. Just my 2 cents

Reply:
Agreed with removing this logic, especially with Alpin's point on our call

https://github.com/osmosis-labs/osmosis/pull/4051/commits/3f4f84d93a1f5d16c4dd22177088668d569ee359

Reply:
When removing this logic though, the following test's usdc balance is off by 100 units instead of just 1 

https://github.com/osmosis-labs/osmosis/blob/3f4f84d93a1f5d16c4dd22177088668d569ee359/x/gamm/keeper/msg_server_test.go#L386-L397

Reply:
I think we need to look into this before merging this. Here is the updated tolerance for the test that is required when reverting this logic

https://github.com/osmosis-labs/osmosis/pull/4051/commits/e0e3a89b8b8a04aa0e5bdda4ffd0488c9c6a3950

Reply:
Let me explain why this tick hack matters for amount1 but not amount0.

When adding print lines to the `msg_server_tests` we see the following for the first three tests:
```
exitCoins: 100000000000eth,100000000000usdc 
amount0: 100000000000 
amount1: 99999999900 

exitCoins: 50000000000eth,50000000000usdc 
amount0: 50000000000 
amount1: 49999999950 

exitCoins: 49999999999eth,49999999999usdc 
amount0: 49999999998 
amount1: 49999999949 
```

ExitCoins is from the ExitPool func from the gamm pool and amount0/amount1 is what actually gets calculated when we create a full range position in the cl pool. As we see, amount0 is always calculating a good number but amount1 is off. This is because the following:

When we run CalcAmount0Delta in cl, we use sqrtPriceCurrent (1) and sqrtPriceMax (10000000000000000000.000000000000000000). From there,
`diff = 9999999999999999999.000000000000000000`
`denominator = 10000000000000000000.000000000000000000`
`liquidity * diff / denominator = amount0 = 100000000000`

When we CalcAmount1Delta however, we us sqrtPriceCurrent (1) and sqrtPriceMin (0.000000001). From there,
`diff = 0.999999999000000000`
`liquidity * diff = amount1 = 99999999900`

Reply:
So it makes sense to me that when we exit we get coins `C`, at ratio `R`, and when we rejoin we join coins `C'` at ratio `R'`, and `C'` makes full use of exactly one of the two coins.

What would make sense to me to happen, is to be single-asset joining the remaining `C - C'` coins.

Reply:
Thanks Dev, this actually makes sense. I think right now I am going to leave the remaining dust in the account holders wallet instead of doing a second join for the sake of simplicity / gas. We can revisit later, but yeah this seems like a non issue now that you have explained it this way.